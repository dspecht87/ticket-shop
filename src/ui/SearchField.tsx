/* eslint-disable @typescript-eslint/ban-types */
import { styled } from "@mui/material/styles";
import { InputAdornment, TextField } from "@mui/material";

const StyledTextField = styled(TextField)({
  marginTop: 5,
  height: 50,
  "& .MuiInputBase-root": {
    backgroundColor: "#F2F2F2",
    borderRadius: 6,
  },
  "& fieldset": { border: "none" },
  "& .MuiInputBase-input": {
    padding: 8,
  },
});

interface Props {
  center: boolean;
  id: string;
  value?: string;
  onChange?: Function;
}

const SearchField: React.FC<Props> = ({
  center,
  id,
  onChange = () => {},
  ...rest
}) => {
  return (
    <div style={{ maxWidth: 312, marginTop: center ? 5 : 0 }}>
      <StyledTextField
        id={id}
        variant="outlined"
        onChange={(event) => onChange(event?.target?.value)}
        fullWidth
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <span
                className="material-symbols-rounded"
                style={{ color: "black" }}
              >
                search
              </span>
            </InputAdornment>
          ),
          // TODO: icon with shorter handle
        }}
        {...rest}
      />
    </div>
  );
};

export default SearchField;
