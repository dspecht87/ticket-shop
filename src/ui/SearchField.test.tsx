import "@testing-library/jest-dom";
import * as renderer from "react-test-renderer";
import SearchField from "./SearchField";

describe("SearchField", () => {
  it("renders correctly with default values", () => {
    const tree = renderer.create(<SearchField />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
