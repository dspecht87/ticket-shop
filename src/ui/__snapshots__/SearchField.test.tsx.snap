// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders correctly with default values 1`] = `
<div
  style={
    {
      "marginTop": 0,
      "maxWidth": 312,
    }
  }
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1dsj3nq-MuiFormControl-root-MuiTextField-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
        <span
          className="material-symbols-rounded"
          style={
            {
              "color": "black",
            }
          }
        >
          search
        </span>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`renders correctly with default values 1`] = `
<div
  style={
    {
      "marginTop": 0,
      "maxWidth": 312,
    }
  }
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1dsj3nq-MuiFormControl-root-MuiTextField-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
        <span
          className="material-symbols-rounded"
          style={
            {
              "color": "black",
            }
          }
        >
          search
        </span>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
